### DISCUSSION TITLE
Project - sportyshoes.com - Table Definitions

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/spring-framework#60] [Tasks]
Write the SQL commands to create the following tables in the `sportyshoes` database:
1. `users`
2. `admins`
3. `products`
4. `orders`

Submit the DDL statements to create the aforementioned tables.

**Comment by akshaye98** [comment]
```java
create table users(
    -> userid varchar(10),
    -> username varchar(25),
    -> password varchar(25)
    -> );

 create table admin(
    -> name varchar(25),
    -> password varchar(25)
    -> );

create table products(
    -> productid varchar(10),
    -> productname varchar(25),
    -> msrp varchar(10),
    -> quantityInStock varchar(25),
    -> productVendor varchar(25)
    -> );

 create table orders(
    -> productid varchar(10),
    -> orderid varchar(10),
    -> orderdate DATE,
    -> productQuantity int
    -> );

```

**Comment by VikasAK47** [comment]
```sql

mysql> CREATE TABLE users (
    -> USER_ID INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
    -> USER_NAME CHAR(25),
    -> PASSWORD CHAR(25)
    -> );

mysql> CREATE TABLE admins(
    -> ADMIN_ID INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
    -> ADMIN_NAME CHAR(25),
    -> ADMIN_PASSWORD CHAR(25)
    -> );

mysql> CREATE TABLE products(
    -> PRODUCT_ID INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
    -> PRODUCT_NAME CHAR(25),
    -> PRODUCT_MSRP INT(15),
    -> QUANTITY_IN_STOCK INT(15),
    -> PRODUCT_VENDOR CHAR(25)
    -> );
mysql> CREATE TABLE orders(
    -> PRODUCT_REF_ID INT,
    -> ORDER_ID INT(15),
    -> ORDER_DATE DATETIME,
    -> ORDER_QUANTITY INT(15),
    -> FOREIGN KEY (PRODUCT_REF_ID)
    -> REFERENCES products(PRODUCT_ID)
    -> ON DELETE CASCADE
    -> );

**Reply by Lalchand2510** [reply]
``

CREATE TABLE admins(
    ADMIN_ID INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
    ADMIN_NAME CHAR(25),
     ADMIN_PASSWORD CHAR(25)
     );

CREATE TABLE admins (
  admin_id INT NOT NULL AUTO_INCREMENT,
  admin_name VARCHAR(45) NULL,
  admin_password VARCHAR(45) NULL,
  PRIMARY KEY (admin_id));


CREATE TABLE users (
  user_id INT NOT NULL AUTO_INCREMENT,
  user_name VARCHAR(45) NULL,
  user_email VARCHAR(45) NULL,
  user_password VARCHAR(45) NULL,
  PRIMARY KEY (user_id));


CREATE TABLE orders (
 order_id INT NOT NULL,
 order_date VARCHAR(45) NULL,
  order_quntity INT NULL,
  user_id VARCHAR(45) NULL,
  PRIMARY KEY (order_id));

CREATE TABLE products (
  product_id INT NOT NULL,
  product_name VARCHAR(45) NULL,
  product_msrp DECIMAL(25) NULL,
  product_in_stock INT NULL,
  product_brand VARCHAR(45) NULL,
  PRIMARY KEY (product_id));``

**Comment by smitaspatil** [comment]
use SportyShoes;
 create table admin(Name varchar(20),Password varchar(30));
 create table Customer(user_id int primary key, Name varchar(20),Password varchar(30),email varchar(30),address varchar(30) );
 create table Product(product_id int primary key, product_desc varchar(40),Price double);
 create table Purchase_Order(Order_id int primary key,product_id int,foreign key(product_id) references Product(product_id), product_desc varchar(40),price int, date timestamp, shipping_address varchar(40)); ``

**Comment by kanika880** [comment]
```mysql

create table users (user_id int, name varchar(100), password char(10));

create table admin (admin_id int, name varchar(100), password char(10));

create table Product (Product_id int, Product_name varchar(100), MSRP decimal(10,2), Quantity_in_stock int, Product_vendor varchar(200));

create table Orders (Order_id int, Product_id int, user_id int, Order_date date);

**Comment by HeeraHP** [comment]
create database sportyshoes;
use sportyshoes;
create table users(userId int,userName  varchar(255),password varchar(255));
create table admins(adminId int,adminName  varchar(255),password varchar(255));
create table products(productId int,name varchar(255),MSRP int, quantityInStock int, productVendor varchar(255) );
create table orders(orderId int, date date, productId int, userId int);

**Comment by vishwabujji** [comment]
mysql> create table users(
    -> users_id varchar(20),
    -> users_name varchar(20),
    -> users_gmail varchar(20)
    -> );
Query OK, 0 rows affected (4.20 sec)

mysql> create table admins(
    -> admins_id varchar(20),
    -> admins_name varchar(20),
    -> admins_gmail varchar(20)
    -> );
Query OK, 0 rows affected (1.49 sec)

mysql> create table products(
    -> products_id varchar(20),
    -> products_name varchar(20),
    -> product_companyname varchar(20),
    -> products_description varchar(20)
    -> );
Query OK, 0 rows affected (0.62 sec)

mysql> create table orders(
    -> orders_id varchar(20),
    -> orders_quantity varchar(20),
    -> orders_summary varchar(20),
    -> orders_username varchar(20)
    -);

**Comment by pravallikamutyala** [comment]
CREATE DATABASE sportyshoes;
CREATE TABLE admins (id INT NOT NULL PRIMARY KEY AUTO_INCREMENT, name CHAR(30), password CHAR(15));
CREATE TABLE users (id INT NOT NULL  PRIMARY KEY AUTO_INCREMENT, name VARCHAR(30), email VARCHAR(30), password VARCHAR(15));
CREATE TABLE products (id INT NOT NULL  PRIMARY KEY AUTO_INCREMENT, name VARCHAR(30), msrp DECIMAL(5,2), quantity_in_stock INT, product_vendor VARCHAR(30));
CREATE TABLE orders (id INT NOT NULL  PRIMARY KEY AUTO_INCREMENT, product_id INT, FOREIGN KEY(product_id) REFERENCES products(id), user_id INT, FOREIGN KEY(user_id) REFERENCES users(id), order_date TIMESTAMP);

**Comment by akshaychromule61** [comment]
CREATE TABLE users (
USER_ID INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
USER_NAME CHAR(25),
PASSWORD CHAR(25));
INSERT INTO users (USER_ID, USER_NAME, PASSWORD) 
VALUES(101,'Akshaykumar', 'akshay');


CREATE TABLE admins(
ADMIN_ID INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
ADMIN_NAME CHAR(25),
ADMIN_PASSWORD CHAR(25));
INSERT INTO admins (ADMIN_ID, ADMIN_NAME, ADMIN_PASSWORD) 
VALUES(1011,'advik', 'advik');

CREATE TABLE products(
PRODUCT_ID INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
PRODUCT_NAME CHAR(25),
PRODUCT_MSRP INT(15),
 QUANTITY_IN_STOCK INT(15),
PRODUCT_VENDOR CHAR(25));

CREATE TABLE orders(
PRODUCT_REF_ID INT,
ORDER_ID INT(15),
ORDER_DATE DATETIME,
ORDER_QUANTITY INT(15),
FOREIGN KEY (PRODUCT_REF_ID)
REFERENCES products(PRODUCT_ID)
ON DELETE CASCADE);
